syntax = "proto3";

package orders;


// The order service definition.
service OrderService {
    // Sends a request for user orders
    rpc GetUserOrders (UserOrderRequest) returns (UserOrderResponse);
  }
  
  // The request message containing the user's ID.
  message UserOrderRequest {
    string user_id = 1;
  }
  
  // The response message containing the user's orders.
  message UserOrderResponse {
    repeated Order orders = 1;
  }
  
// Order item representation in gRPC
message OrderItem {
  string item_id = 1;
  int32 quantity = 2;
  float unit_price = 3;
}

// Order representation in gRPC
message Order {
  string id = 1;
  string user_id = 2;
  float total_price = 3;
  string address = 4;
  repeated OrderItem items = 5;
  string order_date = 6; // ISO 8601 formatted date string
}